aws-lc-rs/src/kem.rs

```diff
@@ -63,6 +61,67 @@ use aws_lc::{
 use core::{cmp::Ordering, ptr::null_mut};
 use zeroize::Zeroize;
 
+#[cfg(not(feature = "fips"))]
+pub(crate) mod semistable {
+    #![allow(unused)]
+
+    use super::{Algorithm, AlgorithmId};
+
+    const ML_KEM_512_SHARED_SECRET_LENGTH: usize = 32;
+    const ML_KEM_512_PUBLIC_KEY_LENGTH: usize = 800;
+    const ML_KEM_512_SECRET_KEY_LENGTH: usize = 1632;
+    const ML_KEM_512_CIPHERTEXT_LENGTH: usize = 768;
+
+    const ML_KEM_768_SHARED_SECRET_LENGTH: usize = 32;
+    const ML_KEM_768_PUBLIC_KEY_LENGTH: usize = 1184;
+    const ML_KEM_768_SECRET_KEY_LENGTH: usize = 2400;
+    const ML_KEM_768_CIPHERTEXT_LENGTH: usize = 1088;
+
+    const ML_KEM_1024_SHARED_SECRET_LENGTH: usize = 32;
+    const ML_KEM_1024_PUBLIC_KEY_LENGTH: usize = 1568;
+    const ML_KEM_1024_SECRET_KEY_LENGTH: usize = 3168;
+    const ML_KEM_1024_CIPHERTEXT_LENGTH: usize = 1568;
+
+    /// NIST FIPS 203 ML-KEM-512 algorithm.
+    pub const ML_KEM_512: Algorithm<AlgorithmId> = Algorithm {
+        id: AlgorithmId::MlKem512,
+        decapsulate_key_size: ML_KEM_512_SECRET_KEY_LENGTH,
+        encapsulate_key_size: ML_KEM_512_PUBLIC_KEY_LENGTH,
+        ciphertext_size: ML_KEM_512_CIPHERTEXT_LENGTH,
+        shared_secret_size: ML_KEM_512_SHARED_SECRET_LENGTH,
+    };
+
+    /// NIST FIPS 203 ML-KEM-768 algorithm.
+    pub const ML_KEM_768: Algorithm<AlgorithmId> = Algorithm {
+        id: AlgorithmId::MlKem768,
+        decapsulate_key_size: ML_KEM_768_SECRET_KEY_LENGTH,
+        encapsulate_key_size: ML_KEM_768_PUBLIC_KEY_LENGTH,
+        ciphertext_size: ML_KEM_768_CIPHERTEXT_LENGTH,
+        shared_secret_size: ML_KEM_768_SHARED_SECRET_LENGTH,
+    };
+
+    /// NIST FIPS 203 ML-KEM-1024 algorithm.
+    pub const ML_KEM_1024: Algorithm<AlgorithmId> = Algorithm {
+        id: AlgorithmId::MlKem1024,
+        decapsulate_key_size: ML_KEM_1024_SECRET_KEY_LENGTH,
+        encapsulate_key_size: ML_KEM_1024_PUBLIC_KEY_LENGTH,
+        ciphertext_size: ML_KEM_1024_CIPHERTEXT_LENGTH,
+        shared_secret_size: ML_KEM_1024_SHARED_SECRET_LENGTH,
+    };
+}
+
+#[cfg(feature = "fips")]
+mod missing_nid {
+    pub const NID_MLKEM512: i32 = 988;
+    pub const NID_MLKEM768: i32 = 989;
+    pub const NID_MLKEM1024: i32 = 990;
+}
```

Comment:
Is this to account for any AWS-LC versions that may have the `NID_MLKEM*` define missing?

Reply:
Correct. `aws-lc-fips-sys` is aligned with the AWS-LC-FIPS 2.0.x, which doesn't provide these constants.